src/Avalonia.Base/RelativePoint.cs

```diff
@@ -10,7 +11,10 @@ namespace Avalonia
     /// Defines the reference point units of an <see cref="RelativePoint"/> or 
     /// <see cref="RelativeRect"/>.
     /// </summary>
-    public enum RelativeUnit
+#if !BUILDTASK
+    public
+#endif
+    enum RelativeUnit
```

Comment:
Is there any way to remove the BUILDTASK conditionals? I see these all over the place and it doesn't make sense to me. Why would BUILDTASK require internal for these types and require animators disabled? This seems like conditional code that really shouldn't be there in a well designed project structure.

Reply:
Because build task includes these types as a compile files. And with "public" we would end up with two different assemblies that define same type/types. 

I don't know if there is a nicer solution though.

Reply:
In languages that support static linking it is easier, right now there is no easy way to go around it I think. Animators need to be disabled since that code is not visible for build task at all.

https://github.com/dotnet/roslyn/issues/162 is describing static linking that would have helped in this scenario.

Reply:
1. I get that buildtask uses some of the same classes as Avalonia itself. I just don't understand why buildtask isn't compiled separately and then called from Avalonia as needed. It shouldn't matter if they share the same source at all. My assumption is buildtask is needed to build Avalonia itself though. If this is a service available for Avalonia developers, yea I can see an issue with shared/duplicate classes. I'll have to look into what buildtask actually is doing.
2. Worse-case some custom, automated tooling would solve this as well. It would copy the needed classes by source file and perhaps change the namespace/access-modifiers. That is only if splitting the project apart as in num.1 really isn't possible.

Reply:
Possibly we can go with splitting files into two partial parts: 
1. Colors.Parsing.cs - properties and TryParse
2. Color.cs - everything else

Though not sure if it will work in every case. Probably worth an issue.

Reply:
Yeah, doing this like that is one option. Although this still won't solve manual file inclusion and #if for access modifier.
But creating an issue to track this sounds good.

Reply:
Partial file doesn't need to have access modifier IIRC.
Manual file inclusion isn't such big a problem. Though it can be something like ```Avalonia.Base/**/*.Parsing.cs```

Reply:
Well, creating a partial file is probably more complicated than just leaving two #if conditionals. My thinking was special handling shouldn't really be needed at all. There is clearly some deeper integration between Avalonia and build tasks that I'm unaware of to require this. I would normally expect build tasks could do whatever it wants and it would be separate from Avalonia itself.

Either way, it is clear this isn't a change for this PR. Thanks for the discussion.